python/helpers/requirements.txt

```
@@ -2,7 +2,7 @@ pip>=21.3.1,<22.4.0  # Range maintains py36 support TODO: Review python 3.6 supp
 pip-tools>=6.4.0,<6.9.1  # Range maintains py36 support TODO: Review python 3.6 support in April 2023 (eol ubuntu 18.04)
 flake8==5.0.4
 hashin==0.17.0
-pipenv==2022.4.8
+pipenv==2022.11.5
```

To preserve python 3.6 support you can try:
```suggestion
pipenv>=2022.4.8,<2022.11.11
```

---

It appears we've got a good number of tests that fail using the latest pipenv, and the change to the lockfile format could be problematic, we'll almost have to match pipenv version with the project/repo I think to avoid these flip/flop issues.

---

why would you need to match python version of the project? just run your pipenv with latest python version only?

---

@glensc I think the issue is with markers and resolution right, because if your project is on say python 3.7 and a package you depend on drops support for it, then a lock file generated by python 3.11 is going to be much different than the one generated by 3.7.

---

perhaps install different versions based on python version then?

for python 3.7 install whatever is last pipenv is installable there, and for newer python go on with current version

---

We're dropping python 3.7 support on August 17th:
* #7702